redbot/cogs/audio/core/commands/controller.py

```diff
@@ -697,14 +722,16 @@ async def command_summon(self, ctx: commands.Context):
     @commands.bot_has_permissions(embed_links=True)
     async def command_volume(self, ctx: commands.Context, vol: int = None):
         """Set the volume, 1% - 150%."""
-        dj_enabled = self._dj_status_cache.setdefault(
-            ctx.guild.id, await self.config.guild(ctx.guild).dj_enabled()
-        )
+        guild_data = await self.config.guild(ctx.guild).all()
+        if ctx.guild.id in self._dj_status_cache:
+            dj_enabled = self._dj_status_cache[ctx.guild.id]
+        else:
+            dj_enabled = self._dj_status_cache.setdefault(ctx.guild.id, guild_data["dj_enabled"])
```

Comment:
These are unnecessary changes to the code logic, let's just keep it as similar to the original code as possible since this PR is meant for config optimizations, not moving code around (especially when the reason for this moving is unknown).
```suggestion
        dj_enabled = self._dj_status_cache.setdefault(ctx.guild.id, guild_data["dj_enabled"])
```

Reply:
> These are unnecessary changes to the code logic, let's just keep it as similar to the original code as possible since this PR is meant for config optimizations, not moving code around (especially when the reason for this moving is unknown).

More information about why those changes were made can be found at: https://discord.com/channels/133049272517001216/171665724262055936/902639810294988881

For clarity I will repeat some of what I said in the convo at that link when I partially reviewed these changes in that discussion without commenting on this issue:

___
Instead of 
```
dj_enabled = self._dj_status_cache.setdefault(
            ctx.guild.id, await self.config.guild(ctx.guild).dj_enabled()
        )
```
Change it to:
```
if ctx.guild.id in self._dj_status_cache:
    dj_enabled = self._dj_status_cache[ctx.guild.id]
else:
    dj_enabled = self._dj_status_cache.setdefault(ctx.guild.id, await self.config.guild(ctx.guild).dj_enabled())
```

2 - Keep guild_from_id as is as it serves as a demo example for newcomers and that's what core cogs are supposed to do in my opinion
3 - Be considerate of .all() calls as they can be expensive, especially on larger bots
3.1 - Sometimes 2 config calls is cheaper than an .all() so if a command has early exits that's likely why .all() wasnt used on it... as making an expensive .all() call would be a waste of resources and potentially slow the bot down (like a user spamming [p]repeat when DJ mode is enabled)
___

Jack, if you want changes like the example here done in another PR that's understandable and I don't need to fight against it, this comment is for providing clarity on why changes like this were added to this PR.